test/framework/include/test_framework/models/TxInput.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <mw/common/Macros.h>
+#include <mw/models/tx/Input.h>
+#include <mw/models/wallet/StealthAddress.h>
+#include <mw/crypto/Bulletproofs.h>
+#include <mw/crypto/Random.h>
+#include <mw/crypto/Schnorr.h>
+
+TEST_NAMESPACE
+
+class TxInput
+{
+public:
+    TxInput(const BlindingFactor& blindingFactor, const SecretKey& sender_privkey, const uint64_t amount, const Input& input)
+        : m_blindingFactor(blindingFactor), m_privkey(sender_privkey), m_amount(amount), m_input(input) { }
+
+    static TxInput Create(
+        const BlindingFactor& blindingFactor,
+        const SecretKey& sender_privkey,
+        const uint64_t amount)
+    {
+        Commitment commitment = Crypto::CommitBlinded(
+            amount,
+            blindingFactor
+        );
+        Signature sig = Schnorr::Sign(
+            sender_privkey.data(),
+            InputMessage()
+        );
+
+        return TxInput(
+            blindingFactor,
+            sender_privkey,
+            amount,
+            Input{
+                Crypto::CommitBlinded(amount, blindingFactor),
```

Comment:
Didn't we already calculate this on line 23?

Reply:
Oops, yep. I meant to use the Commitment from line 23